File: HackerRank Interview Prep Kit/Warm Up/Counting Valleys.py
def solution():
    n = int(input())
    arr = str(input())
    count = 0
    temp = 0
    for val in arr:
        if val == "U":
            temp += 1
        if val == "D":
            if temp == 0:
                count += 1
            temp -= 1
            
    return count

if __name__ == '__main__':
    print(solution())

File: HackerRank Interview Prep Kit/Warm Up/counting_valleys.cpp
#include <iostream>

using namespace std;

int main()
{
    int count = 0;
    int num;
    cin >> num;

    string val;
    cin >> val;

    int temp = 0;
    for (auto letter : val)
    {
        if (letter == 'D' && temp == 0)
        {
            count++;
            temp--;
        }
        else if (letter == 'U')
        {
            temp++;
        }
        else if (letter == 'D')
        {
            temp--;
        }
    }
    cout << count;
    return 0;
}